flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -273,7 +282,21 @@ public Collection<Map<String, List<T>>> process(
                 new PriorityQueue<>(NFAState.COMPUTATION_STATE_COMPARATOR);
 
         for (ComputationState computationState : nfaState.getPartialMatches()) {
-            if (isStateTimedOut(computationState, timestamp)) {
+            String currentStateName = computationState.getCurrentStateName();
+            boolean withinStateTimeout =
```

Comment:
The name *withinStateTimeout* and *isStateTimeout* are not quite clear. What about finding some other name, e.g.  isStateTimeoutBetweenEvents vs isStateTimeoutWholePattern